operations/configuration_management/Pre_Boot_NCN_Image_Configuration.md

```diff
@@ -0,0 +1,129 @@
+1. [Create an Image Customization CFS Session](Create_an_Image_Customization_CFS_Session.md)
+
+1. Download the Resultant NCN Artifacts
+
+    **NOTE:** `$IMS_RESULTANT_IMAGE_ID` is the `result_id` returned in the output of the last command
+    in the previous step:
+
+    ```console
+    ncn# cray cfs sessions describe example --format json | jq .status.artifacts
+    ```
+
+    **NOTE:** If `create-ims-initrd.sh` was not run in the CFS session, do *not* download the `initrd` and `kernel`. Only download the SquashFS file.
+
+    ```console
+    ncn# cray artifacts get boot-images $IMS_RESULTANT_IMAGE_ID/rootfs kubernetes-$ARTIFACT_VERSION-1.squashfs
+
+    ncn# cray artifacts get boot-images $IMS_RESULTANT_IMAGE_ID/initrd initrd.img-$ARTIFACT_VERSION-1.xz
+
+    ncn# cray artifacts get boot-images $IMS_RESULTANT_IMAGE_ID/kernel 5.3.18-150300.59.43-default-$ARTIFACT_VERSION-1.kernel
+    ```
+
+1. Upload NCN boot artifacts into S3
+
+    This steps assumes that the `docs-csm` RPM is installed.
+
+    **NOTE:** If `create-ims-initrd.sh` was not run in the CFS session, use the same `initrd` and `kernel` that were retrieved in step 1.
+
+    ```console
+    ncn# /usr/share/doc/csm/scripts/ceph-upload-file-public-read.py --bucket-name ncn-images --key-name "k8s/$ARTIFACT_VERSION-1/filesystem.squashfs" --file-name kubernetes-$ARTIFACT_VERSION-1.squashfs
+
+    ncn# /usr/share/doc/csm/scripts/ceph-upload-file-public-read.py --bucket-name ncn-images --key-name "k8s/$ARTIFACT_VERSION-1/initrd" --file-name initrd.img-$ARTIFACT_VERSION-1.xz
+
+    ncn# /usr/share/doc/csm/scripts/ceph-upload-file-public-read.py --bucket-name ncn-images --key-name "k8s/$ARTIFACT_VERSION-1/kernel" --file-name $ARTIFACT_VERSION-1.kernel
+    ```
+
+1. Update NCN Boot Parameters
+
+    Get the existing `metal.server` setting for the xname of the node of interest:
+
+    ```console
+    ncn# XNAME=<your-xname>
+    ncn# METAL_SERVER=$(cray bss bootparameters list --hosts $XNAME --format json | jq '.[] |."params"' \
+         | awk -F 'metal.server=' '{print $2}' \
+         | awk -F ' ' '{print $1}')
+    ```
+
+    Verify the variable was set correctly: `echo $METAL_SERVER`
+
+    Update the kernel, initrd, and metal server to point to the new artifacts.
+
+    ```console
+    ncn# S3_ARTIFACT_PATH=ncn-images/k8s/$ARTIFACT_VERSION-1
+    ncn# NEW_METAL_SERVER=http://rgw-vip.nmn/$S3_ARTIFACT_PATH
+
+    ncn# PARAMS=$(cray bss bootparameters list --hosts $XNAME --format json | jq '.[] |."params"' | \
+         sed "/metal.server/ s|$METAL_SERVER|$NEW_METAL_SERVER|" | \
+         tr -d \")
+    ```
+
+    Verify the value of `$NEW_METAL_SERVER` was set correctly within the boot parameters: `echo $PARAMS`
+
+    ```console
+    ncn# cray bss bootparameters update --hosts $XNAME   \
+         --kernel "s3://$S3_ARTIFACT_PATH/kernel" \
+         --initrd "s3://$S3_ARTIFACT_PATH/initrd" \
+         --params "$PARAMS"
+    ```
+
+1. [Reboot the NCN](../node_management/Reboot_NCNs.md)
```

Comment:
Think you still need to wipe the bootloader and remove the disk boot entries, at least for GB nodes.